with

```diff
@@ -20,20 +20,20 @@ while [ true ]; do
   while IFS="" read -r -e -d $'\n' -p "$pmpt " options; do
     history -s "$options"
     
-    curr_command="$(echo $options | { read first rest ; echo $first ; })" 
+    curr_command="$({ read first rest ; echo $first ; } <<<"$options")" 
```

Comment:
I'm not swayed either way about the direction of input. What's wrong with echoes?